utils.js

```diff
@@ -0,0 +1,68 @@
+// TODO
+// await use axios post seedData fn to push data to db
+// done: delete file directory(or add to gitignore so zip files doesn’t push to Github) + drain
+
+const fs = require('fs');
+const path = require('path');
+const unzipper = require('unzipper'); // npm lib used to unzip files
+const { createReadStream } = require('fs');
+
+// Async fs reads and unzip files to a directory
+const folderName = 'Archive.zip';
+const pathToExtract = 'unzippedArchive';
+
+const unzipFolder = (zippedFolder, destination) => {
+  fs.createReadStream(`./${zippedFolder}`).pipe(
+    unzipper.Extract({ path: __dirname + `/${destination}` })
+  );
+};
+
+class File {
+  constructor(filename) {
+    this.filename = filename;
+    this.fileContents;
+  }
+
+  // Read a file and save into the constructor
+  readFile() {
+    return new Promise((resolve, reject) => {
+      try {
+        fs.createReadStream(this.filename)
+
+          .on('data', chunk => {
+            this.fileContents = JSON.parse(chunk.toString());
+          })
+
+          .on('error', err => {
+            reject(err);
+          })
+
+          .on('close', () => {
+            resolve(this.fileContents);
+          });
+      } catch (err) {
+        reject();
+      }
+    });
+  }
+
+  // This method is here in the case we wanted to do invoke a function on the this.fileContents
+  async processFile(processFileFunction, ...otherArgs) {
+    const results = processFileFunction(...otherArgs);
+    return results;
+  }
+}
+
+// Used to control the flow of reading a file.
+const processFileController = async fileName => {
```

Comment:
How does this control the flow of reading a file? This function is just a `try..catch` wrapper around the `File` class. If you want devs to call this method instead of the `File` class don't export it.

Reply:
Well, this is not totally complete. After reading a file, there can be other actions on the file, such as formatting and sending it to the database. I think it will make sense when it's complete. But I get what you are saying.

Reply:
Can you add a `TODO` please.